client/auth/request-login-email-form.jsx

```diff
@@ -0,0 +1,174 @@
+		} );
+	};
+
+	handleError = error => {
+		const { translate } = this.props;
+		this.setState( {
+			errorMessage: error.message
+				? error.message
+				: translate( 'Could not request a login email. Please try again later.' ),
+		} );
+	};
+
+	handleNoticeDismiss = () => {
+		this.setState( {
+			errorMessage: null,
+			hasSubmitted: false,
+		} );
+	};
+
+	handleSubmit = event => {
+		event.preventDefault();
+
+		const emailAddress = formState.getFieldValue( this.state.form, 'emailAddress' );
+		debug( 'form submitted!', emailAddress );
+
+		if ( ! emailAddress ) {
+			return;
+		}
+
+		this.setState( {
+			hasSubmitted: true,
+		} );
+
+		wpcom.undocumented().requestMagicLoginEmail( {
+			email: emailAddress
+		}, ( error, data ) => {
+			if ( error ) {
+				this.handleError( error );
+				return;
+			}
+			debug( 'Request successful', data );
+			window.location.replace( '/login/link-was-sent?email=' + encodeURIComponent( emailAddress ) );
+		} );
+	};
+
+	setFormState = state => {
+		debug( 'setFormState', state );
+		this.setState( { form: state } );
+	};
+
+	componentWillMount() {
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'emailAddress' ],
+			debounceWait: 100,
+			validatorFunction: this.validate.bind( this ),
+			onNewState: this.setFormState,
+			hideFieldErrorsOnChange: false,
+			initialState: {
+				emailAddress: {
+					value: '',
+				}
+			}
+		} );
+
+		this.setFormState( this.formStateController.getInitialState() );
+	}
+
+	validate( formValues ) {
+		const emailAddress = formValues.emailAddress;
+		const isEmailValid = emailValidator.validate( emailAddress );
+		debug( 'validate', emailAddress, isEmailValid );
+		this.setState( {
+			isEmailValid
+		} );
+	}
+
+	render() {
+		const { currentUser, translate } = this.props;
+		return (
+			<div>
+				{ this.state.errorMessage &&
+					<Notice
+						duration={ 10000 }
+						text={ this.state.errorMessage }
+						className="auth__request-login-email-form-notice"
+						showDismiss={ true }
+						onDismissClick={ this.handleNoticeDismiss }
+						status="is-error" />
+				}
+				<LoggedOutForm onSubmit={ this.handleSubmit }>
+					<p>{
+						translate( 'Get a link sent to the email address associated ' +
+							'with your account to log in instantly without your password.' )
+					}</p>
+					{ currentUser && currentUser.username &&
+						<p>{
+							translate( 'NOTE: You are already logged in as user: %(user)s', {
+								args: {
+									user: currentUser.username
+								}
+							} ) }</p>
+					}
+					<FormFieldset>
+						<FormTextInput
+							autoCapitalize="off"
+							autoFocus="true"
+							name="emailAddress"
+							placeholder="Email address"
+							value={ this.state.form.emailAddress.value }
+							onChange={ this.handleChangeEvent }
+						/>
+
+						<LoggedOutFormFooter>
+							<FormButton primary disabled={ this.state.hasSubmitted || ! this.state.isEmailValid }>
+								{ translate( 'Request Email' ) }
```

Comment:
I know the copy isn't final yet; just mentioning that I find this button text very confusing :)